lib/xwax/timecoder.c

```diff
@@ -653,24 +732,42 @@ static void process_sample(struct timecoder *tc,
         if (tc->def->flags & SWITCH_PHASE)
 	    forwards = !forwards;
 
+        track_quadrature_phase(tc, forwards != tc->forwards);
+
         if (forwards != tc->forwards) { /* direction has changed */
             tc->forwards = forwards;
             tc->valid_counter = 0;
         }
+
     }
 
-    /* If any axis has been crossed, register movement using the pitch
-     * counters */
+    /*
+     * If any axis has been crossed, register movement using the pitch
+     * counters. This occurs four time per cycle of the sinusoid.
+     */
 
     if (!tc->primary.swapped && !tc->secondary.swapped)
-	pitch_dt_observation(&tc->pitch, 0.0);
+        if (tc->use_legacy_pitch_filter)
+            pitch_dt_observation(&tc->pitch, 0.0);
+        else
+            pitch_kalman_update(&tc->pitch_kalman, 0.0);
     else {
 	double dx;
 
-	dx = 1.0 / tc->def->resolution / 4;
+        /*
+         * Assumption: We usually advance by a quarter rotation,
+         * unless we skip zero crossings. In this case the new quadrature
+         * tracker calculates the correct displacement for the pitch filter.
+         */
+
+	dx = quantize_phase(tc);
 	if (!tc->forwards)
 	    dx = -dx;
-	pitch_dt_observation(&tc->pitch, dx);
+
+        if (tc->use_legacy_pitch_filter)
+            pitch_dt_observation(&tc->pitch, dx);
+        else
+            pitch_kalman_update(&tc->pitch_kalman, dx);
```

Comment:
Is this condition block conditional to `!tc->forwards`? If yes, you need add curly bracket at line 764, otherwise, perhaps remove the indentation

Reply:
yeah it looks like the 764 has a tab, which is rendered as 2 spaces, and 767 is 8 spaces.

Reply:
Mark had his editor misconfigured once. He just recently fixed this after reading that I publicly stated in the MK2 PR that formatting is not consistent in xwax. 

I can can make a small formatting PR. Maybe even add a clang-format file to the xwax folder. Usually this is discouraged in large projects, but in this case it's fine I think.

Reply:
Yeah unfortunately if we do a mass formatting fix it will make merging/diffing with Mark's code harder.  Many code formatters provide the option to only run formatting on changed lines, that might be the thing to do here.  Not blocking for this PR though!

Reply:
I added a kernel-based clang-format file with minor changes and formatted the whole library.